src/evaluation/game_phase.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include "board_defs.h"
+#include "magic_enum/magic_enum.hpp"
+
+#include <algorithm>
+#include <cstdint>
+
+namespace evaluation::gamephase {
+
+namespace {
+
+constexpr uint8_t s_mgLimit { 24 };
+
+struct Weight {
+    uint8_t mg;
+    uint8_t eg;
+};
+
+[[nodiscard]] constexpr Weight computeWeight(uint8_t materialPhase)
+{
+    /* we have to limit the score in case of early promotions */
+    const uint8_t mgPhase = std::min(materialPhase, s_mgLimit);
+    const uint8_t egPhase = s_mgLimit - mgPhase;
+
+    return { mgPhase, egPhase };
```

Comment:
@clarashepherd I see your point, but I kinda prefer isolating the weight handling here. I think that's more readable? Or maybe I'm missing something? Please throw some physics at me! :)

Reply:
I'm not sure we're actually isolating the weight handling here though - we're just producing two numbers that aren't meaningful until they're used in a weighting formula.

This isn't a big point, and I'm also happy with the two weights here. Use your judgement, this isn't a blocker for me :)